src/main/java/com/boardgo/common/utils/SecurityUtils.java

```diff
@@ -0,0 +1,27 @@
+package com.boardgo.common.utils;
+
+import com.boardgo.domain.user.entity.RoleType;
+import com.boardgo.domain.user.service.dto.CustomUserDetails;
+import java.util.Collection;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public abstract class SecurityUtils {
+
+    private static final Authentication authentication =
+            SecurityContextHolder.getContext().getAuthentication();
+
+    public static boolean hasUserRole() {
+        Collection<? extends GrantedAuthority> authorities = authentication.getAuthorities();
+        return authorities.stream()
+                .filter(o -> o.getAuthority().equals(RoleType.USER.getCode()))
+                .findAny()
+                .isPresent();
+    }
+
+    public static Long currentUserId() {
+        CustomUserDetails user = (CustomUserDetails) authentication.getPrincipal();
+        return user.getId();
+    }
+}
```

Comment:
혹시 이 추상클래스는 어디에 쓰는걸까요 ?_?

Reply:
소셜 회원가입 할 때 만들었는데, `currentUserId()`는 회원Id 가져오는 경우가 많을 것같아서 유틸을 따로 만들었어요!
`hasUserRole` 는 아직 사용하는 곳은 없지만 회원/비회원 구분할 때 필요하지 않을까 해요

Reply:
만약 사용한다면 Spring Bean에 상속해서 하는 걸까요 ?

Reply:
아뇨 JwtFilter 에서 인증된 회원일 경우 Security context에 setAuthentication() 로 인증 정보를 설정하고,
SecurityContextHolder 는 전역적으로 접근 가능해서 설정 해 둔 시큐리티 컨텍스트에서 가져와서 사용해요!

Reply:
```suggestion
public abstract class SecurityUtils {

    public static boolean hasUserRole() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null) {
            return false;
        }
        Collection<? extends GrantedAuthority> authorities = authentication.getAuthorities();
        return authorities.stream()
                .anyMatch(o -> o.getAuthority().equals(RoleType.USER.getCode()));
    }

    public static Long currentUserId() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication == null) {
            throw new OAuth2Exception("context에 인증 정보 존재하지 않습니다.");
        }
        if (!(authentication.getPrincipal() instanceof CustomUserDetails)) {
            throw new OAuth2Exception("CustomUserDetails가 아닙니다.");
        }
        CustomUserDetails user = (CustomUserDetails) authentication.getPrincipal();
        return user.getId();
    }
}
```
이쪽이 더 좋아보입니다! 
아무래도 멀티스레딩 환경이니깐 정적 변수는 쓰지 않는게 좋을 것 같네요 :)

Reply:
아하! 그걸 생각못했네요 그러게요 여러 스레드가 동시에 접근하면 인증정보가 혼동될 수도 있겠네요..